### PR TITLE
added jest translation test + removed python translation test

### PR DESCRIPTION
**Pull request by Zoolixx**
[OM-336]

[OM-336]: https://openmotics.atlassian.net/browse/OM-336

### CODE REVIEW

**Comment by Zoolixx** [comment]
@openmotics-ci test frontend=feature/jest-translation-tests gateway=develop

File: tests/I18N.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+describe("the html files", () => {
+    const {readFileSync} = require("fs")
+    const FileSet = require('file-set');
+    var failures = [];
```
Comment (khenderick) [inline tests/I18N.test.js]: Use `let`.

File: tests/I18N.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                for (let match of matches) {
+                    if (match.count("'") % 2) {
+                        console.error(`[!!] File ${file} contains string literal with mismatched quotes`);
+                        failures.push(`${file}.${contents}`);
+                    }
+                }
+            }
+        }
+    }
+    it("should find proper translated html files", () => {
```
Comment (khenderick) [inline tests/I18N.test.js]: Missing space

File: tests/I18N.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        let file_set = new FileSet('src/**/*.html');
+        check_translation_in_html(file_set.files);
+        expect(failures.length).toEqual(0);
+    });
+});
+
+String.prototype.count = function(c) {
+    let result = 0;
+    for (let char of this){
```
Comment (khenderick) [inline tests/I18N.test.js]: Missing space

File: tests/translation.test.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+describe("the translation file", () => {
+    const {readdirSync, statSync} = require("fs")
+    const {join} = require("path")
+    const dirs = p => readdirSync(p).filter(f => statSync(join(p, f)).isDirectory())
+    var langs = dirs("src/locales/")
```
Comment (khenderick) [inline tests/translation.test.js]: Use `let`.

File: tests/translation.test.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+    const check_translation_against_base = function(base, trans, path) {
+
+        Object.keys(base).forEach(key => {
+            let tempPath = `${path}.${key}`;
+            if (key === "icon" || key === "icons") {
+                return;
+            }
+            if (typeof base[key] === "string") {
+                if (!Object.keys(trans).includes(key) || typeof trans[key] != "string" || trans[key].startsWith("TRANSLATE:")) {
```
Comment (khenderick) [inline tests/translation.test.js]: Use `!==` iso `!=`

File: tests/translation.test.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+                    console.error(`[!!] Missing translation ${tempPath}`); // Logging ends up in Jenkins build console
+                    failures.push(tempPath);
+                    return;
+                }
+            } else if (typeof base[key] === "object" && base[key] != null && !(base[key] instanceof Array) && !(base[key] instanceof Date)) {
+
+                if (!Object.keys(trans).includes(key)) {
```
Comment (khenderick) [inline tests/translation.test.js]: Spaces

File: tests/translation.test.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+                return;
+            }
+            if (typeof base[key] === "string") {
+                if (!Object.keys(trans).includes(key) || typeof trans[key] != "string" || trans[key].startsWith("TRANSLATE:")) {
+                    console.error(`[!!] Missing translation ${tempPath}`); // Logging ends up in Jenkins build console
+                    failures.push(tempPath);
+                    return;
+                }
+            } else if (typeof base[key] === "object" && base[key] != null && !(base[key] instanceof Array) && !(base[key] instanceof Date)) {
+
```
Comment (khenderick) [inline tests/translation.test.js]: Spacing

File: tests/I18N.test.js (hunk 2 above)
Comment (Zoolixx) [reply tests/I18N.test.js]: Not clear

Comment (khenderick) [reply tests/I18N.test.js]: Sorry, nevermind, it's not an `if` :confused:

File: tests/translation.test.js [hunk 8]
```diff
@@ -29,18 +29,18 @@ describe("the translation file", () => {
...
                 }
-            } else if (typeof base[key] === "object" && base[key] != null && !(base[key] instanceof Array) && !(base[key] instanceof Date)) {
+            }else if (typeof base[key] === "object" && base[key] !== null && !(base[key] instanceof Array) && !(base[key] instanceof Date)) {
 
                 if (!Object.keys(trans).includes(key)) {
                     console.error(`[!!] Missing translation ${tempPath}`);
                     failures.push(tempPath);
                     return;
-                } else if (typeof base[key] != "object") {
+                }else if (typeof base[key] !== "object") {
```
Comment (khenderick) [inline tests/translation.test.js]: You actually removed a space that was in the correct location?

File: tests/translation.test.js [hunk 9]
```diff
@@ -29,18 +29,18 @@ describe("the translation file", () => {
...
             }
             if (typeof base[key] === "string") {
-                if (!Object.keys(trans).includes(key) || typeof trans[key] != "string" || trans[key].startsWith("TRANSLATE:")) {
+                if (!Object.keys(trans).includes(key) || typeof trans[key] !== "string" || trans[key].startsWith("TRANSLATE:")) {
                     console.error(`[!!] Missing translation ${tempPath}`); // Logging ends up in Jenkins build console
                     failures.push(tempPath);
                     return;
                 }
-            } else if (typeof base[key] === "object" && base[key] != null && !(base[key] instanceof Array) && !(base[key] instanceof Date)) {
+            }else if (typeof base[key] === "object" && base[key] !== null && !(base[key] instanceof Array) && !(base[key] instanceof Date)) {
```
Comment (khenderick) [inline tests/translation.test.js]: Why did you remove this space again?

File: tests/translation.test.js (hunk 7 above)
Comment (khenderick) [reply tests/translation.test.js]: Still some messed-up white space here.
